core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -1027,6 +1027,28 @@ private void alterTableDropColumn(int tableNamePosition, TableWriter writer) thr
     }
 
     private void alterTableDropPartition(TableWriter writer) throws SqlException {
+        CharSequence tok = expectToken(lexer, "'list' or 'where'");
+        if (SqlKeywords.isListKeyword(tok)) {
+            alterTableDropPartitionByList(writer);
+        } else if (SqlKeywords.isWhereKeyword(tok)) {
+            ExpressionNode expr = parser.expr(lexer, (QueryModel) null);
+            int timestampIndex = writer.getMetadata().getTimestampIndex();
+            if (timestampIndex != -1) {
+                String timestampColumnName = writer.getMetadata().getColumnName(timestampIndex);
+                Function function = functionParser.parseFunction(expr, new SingleTimestampColumnRecordMetadata(timestampColumnName), currentExecutionContext);
```

Comment:
such metadata can be cached with writer to reduce allocations. When we cache this with writer we would need to check what happens when timestamp column is renamed, can we still drop partitions?